sentry-core/src/main/java/io/sentry/core/SentryEnvelope.java

```diff
@@ -1,3 +1,34 @@
 package io.sentry.core;
 
-public class SentryEnvelope {}
+import io.sentry.core.protocol.SentryId;
+import io.sentry.core.util.Nullable;
+
+public class SentryEnvelope {
+
+  private final SentryEnvelopeHeader header;
+  private final Iterable<SentryEnvelopeItem> items;
```

Comment:
Values are parsed already. It's just a list of header items.
How would stream fit here?

Reply:
if I'm not wrong, `Iterable` has been added to Java 8 to support `forEach(Consumer<? super T> action)` and `Consumer/Java streams` are not backported yet to all Android API levels.
So I'd only use a `List<T>/ArrayList<T>` instead of `Iterable` to keep the compatibility with all our other list-objects.

Reply:
Seems like `Iterable` is API 1, only the `.forEach` on it is API 24:
https://developer.android.com/reference/java/lang/Iterable#forEach(java.util.function.Consumer%3C?%20super%20T%3E)

I'd suggest we stick with `Iterable`s then, anyway the tooling won't let us use `iterable.forEach` on it.
Agreed?

Reply:
okey, agreed